sqlglot/expressions.py

```diff
@@ -2012,3 +2014,37 @@ def not_(expression, dialect=None, **opts):
         **opts,
     )
     return Not(this=_wrap_operator(this))
+
+
+SAFE_IDENTIFIER_RE = re.compile(r"^[a-zA-Z][\w]*$")
+
+
+def alias(expression, alias, dialect=None, **opts):
+    """
+    Create an Alias expression.
+    Expample:
+        >>> alias('foo', 'bar').sql()
+        'foo AS bar'
+
+    Args:
+        expression (str or Expression): the SQL code strings to parse.
+            If an Expression instance is passed, this is used as-is.
+        alias (str or Identifier): the alias name to use. If the name has
+            special charachters it is quoted.
+        dialect (str): the dialect used to parse the input expression.
+        **opts: other options to use to parse the input expressions.
+
+    Returns:
+        Alias: the aliased expression
+    """
+    exp = _maybe_parse(expression, dialect=dialect, parser_opts=opts)
+    if isinstance(alias, Identifier):
```

Comment:
`SAFE_IDENTIFIER_RE` feels like parser logic.

I think you can replace these few lines with:

```python
identifier = _maybe_parse(alias, parse_into=Identifier, dialect=dialect, parser_opts=opts)
```

We might also want to add some kind of check in `_maybe_parse`:
```
if isinstance(code_or_expression, Expression):
    if parse_into and not isinstance(code_or_expression, parse_into):
        raise ValueError(...)
    return code_or_expression
```
And add `Identifier` to the map here: https://github.com/tobymao/sqlglot/blob/main/sqlglot/parser.py#L261

Reply:
Er, I think you can alias things that aren't identifiers?

e.g.
```SUM(foo) AS bar``

What are we trying to accomplish here? I'm assuming:

```python
select(alias("foo", "bar")).from("baz")
```
?

Reply:
you can't parse_into an identifier for example

parse_one("x-1") you want this to be an identifier "x-1", but sqlglot doesn't know you intended it to be a string